WcaOnRails/app/controllers/api/v0/competitions_controller.rb

```diff
@@ -107,13 +107,14 @@ def show_wcif
 
   def show_wcif_public
     id = params[:competition_id] || params[:id]
-
     cache_key = "wcif/#{id}"
+    competition = competition_from_params
     expires_in 5.minutes, public: true
-    render json: Rails.cache.fetch(cache_key, expires_in: 5.minutes) {
-      competition = competition_from_params
-      competition.to_wcif
-    }
+    if stale?(last_modified: competition.updated_at, public: true)
```

Comment:
https://apidock.com/rails/ActionController/ConditionalGet/stale%3F specifies that you can simply pass the competition model itself and Rails is smart enough to figure out that it has an `updated_at` flag that can be used. Also gives you `etag` for free.

Reply:
Also, I don't really understand the code logic behind this. You use an `if` check to figure out whether you should return something. But what if that `if` check yields `false`? What does the controller return in that case?

Reply:
Reading the documentation, it seems as though Rails sends a `304` response if the resource is considered stale. Are we confident that our existing API users can handle this kind of response?